front/lib/models/assistant/agent.ts

```diff
@@ -401,3 +369,297 @@ AgentUserRelation.belongsTo(User, {
+  foreignKey: { name: "retrievalConfigurationId", allowNull: true }, // null = no retrieval action set for this Agent
+});
+AgentConfiguration.belongsTo(AgentRetrievalConfiguration, {
+  as: "retrievalConfiguration",
+  foreignKey: { name: "retrievalConfigurationId", allowNull: true }, // null = no retrieval action set for this Agent
+});
+
+// NEW -- AgentConfig -> RetrievalConfig (1:N)
+AgentConfiguration.hasMany(AgentRetrievalConfiguration, {
+  // TODO(@fontanierh) make it non-nullable
+  foreignKey: { name: "agentConfigurationId", allowNull: true },
+  onDelete: "CASCADE",
+});
+AgentRetrievalConfiguration.belongsTo(AgentConfiguration, {
+  // TODO(@fontanierh) make it non-nullable
+  foreignKey: { name: "agentConfigurationId", allowNull: true },
+  onDelete: "CASCADE",
+});
+
+// TODO(@fontanierh) TO BE MOVED TO THE dust_app_run.ts file -- inlined during multi actions migration
+// to avoid circular dependencies
+
+export class AgentDustAppRunConfiguration extends Model<
+  InferAttributes<AgentDustAppRunConfiguration>,
+  InferCreationAttributes<AgentDustAppRunConfiguration>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare agentConfigurationId: ForeignKey<AgentConfiguration["id"] | null>;
+
+  declare sId: string;
+
+  declare appWorkspaceId: string;
+  declare appId: string;
+}
+
+AgentDustAppRunConfiguration.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    sId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    appWorkspaceId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    appId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+  },
+  {
+    modelName: "agent_dust_app_run_configuration",
+    indexes: [
+      {
+        unique: true,
+        fields: ["sId"],
+      },
+    ],
+    sequelize: frontSequelize,
+  }
+);
+
+// DEPERECATED -- AgentConfig -> DustAppRunConfig (1:1)
+AgentDustAppRunConfiguration.hasOne(AgentConfiguration, {
+  as: "dustAppRunConfiguration",
+  foreignKey: { name: "dustAppRunConfigurationId", allowNull: true }, // null = no DustAppRun action set for this Agent
+});
+AgentConfiguration.belongsTo(AgentDustAppRunConfiguration, {
+  as: "dustAppRunConfiguration",
+  foreignKey: { name: "dustAppRunConfigurationId", allowNull: true }, // null = no DustAppRun action set for this Agent
+});
+
+// NEW -- AgentConfig -> DustAppRunConfig (1:N)
+AgentConfiguration.hasMany(AgentDustAppRunConfiguration, {
+  // TODO(@fontanierh): make it non-nullable
+  foreignKey: { name: "agentConfigurationId", allowNull: true },
+  onDelete: "CASCADE",
```

Comment:
Is the purpose of CASCADE delete to not break the deletion of agent configuration on account downgrade / GDPR delete?

Reply:
Does it work?

Reply:
It will be useful once we don't allow null. You delete the config, it deletes the ADARC